model-server/src/test/kotlin/org/modelix/model/server/handlers/RepositoriesManagerTest.kt

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.modelix.model.server.handlers
+
+import kotlinx.coroutines.flow.count
+import kotlinx.coroutines.runBlocking
+import org.modelix.model.api.IConceptReference
+import org.modelix.model.api.INode
+import org.modelix.model.api.TreePointer
+import org.modelix.model.api.getRootNode
+import org.modelix.model.lazy.BranchReference
+import org.modelix.model.lazy.CLTree
+import org.modelix.model.lazy.CLVersion
+import org.modelix.model.lazy.RepositoryId
+import org.modelix.model.operations.OTBranch
+import org.modelix.model.server.store.InMemoryStoreClient
+import org.modelix.model.server.store.LocalModelClient
+import kotlin.test.Test
+import kotlin.test.assertNotEquals
+
+class RepositoriesManagerTest {
+
+    @Test
+    fun `delta objects flow does not fail when values are emitted faster than consumed`() = runBlocking {
+        // Arrange
+        val repositoryId = RepositoryId("repo1")
+        val branchRef = repositoryId.getBranchReference()
+        val repositoriesManager = RepositoriesManager(LocalModelClient(InMemoryStoreClient()))
+        repositoriesManager.createRepository(repositoryId, null)
+        repositoriesManager.runWrite(branchRef) { root ->
+            // Creating many children makes the flow emitting many values at once.
+            repeat(100) {
+                root.addNewChild("aChild", -1, null as IConceptReference?)
+            }
+        }
+        val latestVersionHash = repositoriesManager.getVersionHash(branchRef)!!
+
+        // Act / Assert
+        val flow = repositoriesManager.computeDelta(latestVersionHash, null)
+        // `count` is an example for consuming the flow.
+        // The flow should not fail when the consumer is slow.
+        // With the given data, the consumption is slower than the production of values.
+        // Still, the following code should run successfully.
+        val count = flow.count()
```

Comment:
When I run the test without the fix it still succeeds. `.count()` is not sufficient.

Reply:
Hm, ok.
I extended the existing `ModelClientV2Test.kt` instead.
It is one test less and tests less internal things.

Reply:
The modified test fails with without the fix.